src/main/java/com/basho/riak/client/core/RiakNode.java

```diff
@@ -874,6 +898,25 @@ public void onException(Channel channel, final Throwable t)
         }
     }
 
+    private void checkNetworkAddressCacheSettings()
+    {
+        final String property = Security.getProperty("networkaddress.cache.ttl");
+        if (property == null)
+        {
+            return;
+        }
+
+        int cacheTTL = Integer.parseInt(property);
+
+        if (cacheTTL == -1)
```

Comment:
UGH disregard my suggestion. I don't think we should change JVM settings for the user, ever.